[math] FFT test fails with FFTW

FFT with backend fftw fails with C2C transform. 
AFAIK nothing was changed since version 0.12. The only thing I see is numpy having changed its internal FFT backend ([pocketFFT](https://numpy.org/devdocs/release/1.17.0-notes.html)) since `1.17.0`, but then tests should fail with other backends (not only fftw).

Anyway the metric is Maximum Absolute Difference (`max_i |x_i - y_i|`). 
It is not surprising to see `MAD = 0.3` on float32 precision, I wonder why we did not have this issue before.

```bash
======================================================================
FAIL: test_fftw (silx.math.fft.test.test_fft.TestFFT) (trdim='2D', mode='C2C', size=(512, 512))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 144, in __run_tests
    self.__test(backend, trdim, mode, size, **extra_args)
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 182, in __test
    "FFT %s:%s, MAE(%s, numpy) = %f" % (mode, trdim, backend, mae)
AssertionError: False is not true : FFT C2C:2D, MAE(fftw, numpy) = 0.302643

======================================================================
FAIL: test_fftw (silx.math.fft.test.test_fft.TestFFT) (trdim='2D', mode='C2C', size=(512, 511))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 144, in __run_tests
    self.__test(backend, trdim, mode, size, **extra_args)
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 182, in __test
    "FFT %s:%s, MAE(%s, numpy) = %f" % (mode, trdim, backend, mae)
AssertionError: False is not true : FFT C2C:2D, MAE(fftw, numpy) = 0.302643

======================================================================
FAIL: test_fftw (silx.math.fft.test.test_fft.TestFFT) (trdim='2D', mode='C2C', size=(511, 512))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 144, in __run_tests
    self.__test(backend, trdim, mode, size, **extra_args)
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 182, in __test
    "FFT %s:%s, MAE(%s, numpy) = %f" % (mode, trdim, backend, mae)
AssertionError: False is not true : FFT C2C:2D, MAE(fftw, numpy) = 0.302643

======================================================================
FAIL: test_fftw (silx.math.fft.test.test_fft.TestFFT) (trdim='2D', mode='C2C', size=(511, 511))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 144, in __run_tests
    self.__test(backend, trdim, mode, size, **extra_args)
  File "/users/kieffer/workspace-400/silx/build/lib.linux-x86_64-3.7/silx/math/fft/test/test_fft.py", line 182, in __test
    "FFT %s:%s, MAE(%s, numpy) = %f" % (mode, trdim, backend, mae)
AssertionError: False is not true : FFT C2C:2D, MAE(fftw, numpy) = 0.302643
```

--- a/silx/math/fft/test/test_fft.py
+++ b/silx/math/fft/test/test_fft.py
@@ -60,10 +60,10 @@ class TransformInfos(object):
             "C2C_double": np.complex128,
         }
         self.sizes = {
-            "1D": [(512,), (511,)],
-            "2D": [(512, 512), (512, 511), (511, 512), (511, 511)],
-            "3D": [(128, 128, 128), (128, 128, 127), (128, 127, 128), (127, 128, 128),
-                 (128, 127, 127), (127, 128, 127), (127, 127, 128), (127, 127, 127)]
+            "1D": [(128,), (127,)],
+            "2D": [(128, 128), (128, 127), (127, 128), (127, 127)],
+            "3D": [(64, 64, 64), (64, 64, 63), (64, 63, 64), (63, 64, 64),
+                 (64, 63, 63), (63, 64, 63), (63, 63, 64), (63, 63, 63)]
         }
         self.axes = {
             "1D": None,
@@ -80,7 +80,7 @@ class TestData(object):
     def __init__(self):
         self.data = ascent().astype("float32")
         self.data1d = self.data[:, 0]  # non-contiguous data
-        self.data3d = np.tile(self.data[:128, :128], (128, 1, 1))
+        self.data3d = np.tile(self.data[:64, :64], (64, 1, 1))
         self.data_refs = {
             1: self.data1d,
             2: self.data,

